jib-gradle/jib-gradle.yaml

```diff
@@ -36,28 +36,35 @@ spec:
                   project.apply plugin: com.google.cloud.tools.jib.gradle.JibPlugin
                 }
               }
-            }" > /tekton/home/init-script.gradle
+            }" > $HOME/init-script.gradle
       # Runs the Gradle Jib build.
       gradle jib \
         --stacktrace --console=plain \
-        --init-script=/tekton/home/init-script.gradle \
-        -Duser.home=/tekton/home \
-        -Dgradle.user.home=/tekton/home/.gradle \
+        --init-script=$HOME/init-script.gradle \
+        --gradle-user-home=$HOME/.gradle \
+        -Dgradle.user.home=$HOME/.gradle \
+        -Duser.home=$HOME \
         -Djib.allowInsecureRegistries=$(params.INSECUREREGISTRY) \
         -Djib.to.image=$(resources.outputs.image.url)
     workingDir: $(workspaces.source.path)/$(params.DIRECTORY)
+    env:
+    - name: HOME
+      value: /workspace
```

Comment:
any information on why `/workspace`?

and what is the `${workspaces.source.path}` in workingDir (line 49)?

Reply:
`/workspace` (the filesystem directory) is the directory that the current Tekton releases use as a general space for work. Tekton docs do specify this directory in [their examples](https://github.com/tektoncd/pipeline/blob/master/docs/migrating-v1alpha1-to-v1beta1.md#replacing-pipelineresources-with-tasks).

`source` in `$(workspaces.source.path)` is the name defined in line 17. The `source` is a "Workspace" concept in Tekton, and `$(workspaces.source.path)` evaluates to the path of this "Workspace". In our `Task` definition, this "source workspace" is where this `Task` should find Java source to containerize, hence setting the current working directory to it. In practice, this directory is `/workspace/source`.